### ISSUE TITLE
Smooth L1 Loss usage

### ISSUE THREAD

**Issue by leonardoaraujosantos** [weiliu89/caffe#428]
Hi @weiliu89, while trying to understand the reason of using Smooth L1 Loss, I've created this notebook
[SmoothL1Loss.pdf](https://github.com/weiliu89/caffe/files/764291/SmoothL1Loss.pdf)

The reasons that I've found about using the Smooth L1 loss are those:
* Always have a defined gradient
* Regression Loss will be less sensible to small variations between the bonding box on the training dataset and the bounding box predicted during training.

### L1 distance (Manhattan distance)
![l1distance](https://cloud.githubusercontent.com/assets/898383/22792564/1c0e0f1e-eee5-11e6-973c-669a410f8b76.png)

### Gradients SmoothL1 and L1
![gradients](https://cloud.githubusercontent.com/assets/898383/22792734/a45b92f6-eee5-11e6-8e0f-32264c5e7c47.png)

### Questions
1. Did you also tried the normal L1 loss for box regression?

**Comment by weiliu89** [comment]
No. I didn't try L1 loss.

**Comment by leonardoaraujosantos** [comment]
Cool @weiliu89, but do you think that my findings are correct, with respect to SSD multibox loss?

**Comment by wk910930** [comment]
@leonardoaraujosantos Your findings looks good. I like it.

**Comment by leonardoaraujosantos** [comment]
Thanks @wk910930 I'm trying to write a book on gitbook to cover all the stuff that me and my collegues are learning about deep learning (Including the part on SSD for object detection).
https://leonardoaraujosantos.gitbooks.io/artificial-inteligence/content/
Hope it helps.
